src/AnalisadorSintatico.java

```diff
@@ -7,11 +7,12 @@ public AnalisadorSintatico() {
 	}
 	
 	public boolean analisar(String linha, List<String> expressoes) {
-		for(int i =0;i<expressoes.size();i++) {
-			if( linha.trim().matches(expressoes.get(i))) {
+		
+		for (String expressao : expressoes) {
```

Comment:
Esta é uma refatoração de código opcional que torna o código mais legível para aqueles que conhecem a estrutura `for-each`.